File: src/main/java/it/giaquinto/springberry/model/system/SystemExit.java
package it.giaquinto.springberry.model.system;

public enum SystemExit {

    TRIGGERED,
    FINISHED,
    REQUESTED,

    DEFAULT,

    ABNORMAL,
    CRITICAL,
    EXCEPTION,
    ERROR;

    public static int valueOf(final SystemExit systemExit) {
        return switch (systemExit) {
            case TRIGGERED -> 5;
            case FINISHED -> 2;
            case REQUESTED -> 1;
            case DEFAULT -> 0;
            case ABNORMAL -> -1;
            case CRITICAL -> -2;
            case EXCEPTION -> -5;
            case ERROR -> -10;
        };
    }
}

File: src/main/java/it/giaquinto/springberry/model/system/SystemExit.kt
package it.giaquinto.springberry.model.system

enum class SystemExit {
    TRIGGERED, FINISHED, REQUESTED, DEFAULT, ABNORMAL, CRITICAL, EXCEPTION, ERROR;

    companion object {
        fun valueOf(systemExit: SystemExit): Int =
            when (systemExit) {
                TRIGGERED -> 5
                FINISHED -> 2
                REQUESTED -> 1
                DEFAULT -> 0
                ABNORMAL -> -1
                CRITICAL -> -2
                EXCEPTION -> -5
                ERROR -> -10
            }
    }
}